src/System.Net.Http/src/System/Net/Http/HttpClientHandler.cs

```diff
@@ -26,6 +27,11 @@ private static bool UseManagedHandler
                     return true;
                 }
 
+                if (AppContext.TryGetSwitch(AppCtxManagedHandlerSettingName, out bool isManagedEnabled) && isManagedEnabled)
```

Comment:
What's the desired behavior if the environment variable is set to true but the appcontext is explicitly set to false?

Reply:
I would hope we'd take a consistent approach here as we did with the Globalization.Invariant switch.  If the `AppContext` switch is set, respect it.  If it isn't set, check the environment variable.  If we have consistent policies across these settings, it should make it a lot easier on users to understand behavior.

See https://github.com/dotnet/coreclr/blob/b60c91cdb9de36d2356ee8f135de65473f282209/src/mscorlib/src/System/Globalization/GlobalizationMode.cs#L11-L28

Reply:
>  would hope we'd take a consistent approach here 

Yes, we should be consistent, in naming, in order of operations, etc.  @tarekgh, @eerhardt, @AlexGhiondea, is the guidance documented anywhere?

Reply:
I'm not sure if it is documented outside of the comments in the [above linked source code](https://github.com/dotnet/coreclr/blob/b60c91cdb9de36d2356ee8f135de65473f282209/src/mscorlib/src/System/Globalization/GlobalizationMode.cs#L11-L28).  Maybe the others know.

Reply:
If it's not, that should be fixed.

Reply:
I don't think every context switch should be supported by the environment variable. We did that with Globalization invariant switch mainly to support docker images. so I think this is a special case. 

if we need to have environment variable support with all switches, I would say this should be done in AppContext and not the code consuming the switch.

Reply:
the environment variable already exist (COMPlus_UseManagedHttpClientHandler) and it was implemented before this. We can probably still remove it but that would be bigger change as our current unit tests do depend on it.

Reply:
> We can probably still remove it but that would be bigger change as our current unit tests do depend on it.

It should be trivial to change the unit tests at the same time.  Search and replace.

Reply:
Also the name ```COMPlus_UseManagedHttpClientHandler``` is not conforming with the environment variable naming in net core (which prefixed with DOTNET_)

Reply:
yes, it should be easy to change @stephentoub I was not sure if others are aware of it.
I don't have personal prescreens.  Consistent and reasonably long would be nice.